iac-common/src/main/java/org/sonar/iac/common/yaml/object/ListObject.java

```diff
@@ -62,13 +65,17 @@ public Stream<YamlTree> getItemIf(Predicate<YamlTree> predicate) {
   }
 
   public ListObject reportIfAnyItem(Predicate<YamlTree> predicate, String message) {
-    getItemIf(predicate).findFirst().ifPresent(item -> report(message));
+    getItemIf(predicate).findFirst().ifPresent(item -> reportOnItems(message));
```

Comment:
Shouldn't it highlight only on matching items then?

Reply:
The current logic was that it matches all items, e.g. all Capabilities are sensitive. I guess such a method may be needed in the future.